src/realm.h

```diff
@@ -1981,6 +2016,20 @@ RLM_API bool realm_dictionary_get(const realm_dictionary_t*, size_t index, realm
 RLM_API bool realm_dictionary_insert(realm_dictionary_t*, realm_value_t key, realm_value_t value, size_t* out_index,
                                      bool* out_inserted);
 
+/**
+ * Insert an embedded object.
+ *
+ * @return A non-NULL pointer if the object was created successfully.
+ */
+RLM_API realm_object_t* realm_dictionary_insert_embedded(realm_dictionary_t*, realm_value_t key);
```

Comment:
Probably, mention that the ptr returned must be freed by the caller via `real_release()`

Reply:
I think it is convention that any `realm_x` pointer returned is owned by the caller and hence requires an explicit releases, so don't think we mention it on every call. 

`realm_x` pointers available from callback are owned by the callee and has to be `realm_clone`d to transfer ownership/extend lifetime past the duration callback.